src/main/java/spoon/reflect/visitor/chain/CtQuery.java

```diff
@@ -71,4 +72,23 @@
 	 * @return the list of elements collected by the query.
 	 */
 	<R> List<R> list(Class<R> itemClass);
+
+	/**
+	 * Defines whether this query will throw {@link ClassCastException}
+	 * when the output of the previous step cannot be cast to type of input of next step.
+	 * The default value is {@link QueryFailurePolicy#FAIL}<br>
+	 *
+	 * Note: The {@link CtQueryable#filterChildren(Filter)} step never throws {@link ClassCastException}
+	 *
+	 * @param policy the policy
+	 * @return this to support fluent API
+	 */
+	CtQuery failurePolicy(QueryFailurePolicy policy);
+
+	/**
+	 * Sets the name of current query, to identify the current step during debugging of a query

```

Comment:
Do you see different between query and query step? The `name` method sets the name of last "query step".